### PR TITLE
Set base box for parallels environment to parallels/ubuntu-1404

### PR DESCRIPTION
**Pull request by timstallmann**
This PR adds support for parallels by conditionally enabling the parallels Ubuntu 14.04 base box instead of the Virtualbox equivalent.

### CODE REVIEW

**Comment by lhridley** [comment]
Let's create a separate branch for this on the main repo, so that the Parallels build has its own Vagrantfile.  Call it "parallels", or something similar so that it can be differentiated from the Virtualbox branch (we can create a VirtualBox branch also).

**Comment by timstallmann** [comment]
You can set the default virtual environment provider via an environment variable or specify one explicitly via a command-line flag (`vagrant up --provider=parallels`). Theoretically, the vagrant file should "just work" with either base box, but apparently the parallels ubuntu 14.04 base box did not have `curl` or `add-apt-repository` installed so I had to add a conditional step to do that.

File: Vagrantfile [hunk 1]
```diff
@@ -139,8 +156,6 @@ SCRIPT
   #Making box as small as possible after provisioning
   $script = <<SCRIPT
 sudo apt-get clean
-sudo dd if=/dev/zero of=/EMPTY bs=1M
-sudo rm -f /EMPTY
 cat /dev/null > ~/.bash_history && history -c && exit
```
Comment (lhridley) [inline Vagrantfile]: Can we add these two lines back, and wrap them in an if statement, similar to what you did on lines 40-42?

Comment (timstallmann) [inline Vagrantfile]: @lhridley -- done!

**Pull request merged**